src/select.coffee

```diff
@@ -354,14 +386,23 @@ Ember.AddeparMixins.ResizeHandlerMixin,
   deletePressed: Ember.K
 
   escapePressed: (event) ->
-    @send 'hideDropdown'
+    if @get('showDropdown')
+      @send 'hideDropdown'
+      @$()?.focus()
```

Comment:
I see.

It's probably cleaner to check the state of the component ( `if @get('isDestroyed') or @get('isDestroying')` ) at the beginning of the event handler